kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/ExtendedEMFBusinessProcessTest.java

```diff
@@ -0,0 +1,116 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.springboot.samples;
+
+import static org.appformer.maven.integration.MavenRepository.getMavenRepository;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.io.File;
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.persistence.EntityManager;
+import javax.persistence.EntityManagerFactory;
+
+import org.appformer.maven.integration.MavenRepository;
+import org.jbpm.kie.services.impl.KModuleDeploymentUnit;
+import org.jbpm.services.api.DeploymentService;
+import org.jbpm.services.api.ProcessService;
+import org.jbpm.springboot.samples.entities.Person;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.kie.api.KieServices;
+import org.kie.api.builder.ReleaseId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.test.context.TestPropertySource;
+import org.springframework.test.context.junit4.SpringJUnit4ClassRunner;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@SpringBootTest(classes = {JBPMApplication.class, ExtendedEntityManagerFactory.class}, webEnvironment = WebEnvironment.RANDOM_PORT)
+@TestPropertySource(locations="classpath:application-test.properties")
+public class ExtendedEMFBusinessProcessTest {
+    
+    static final String ARTIFACT_ID = "evaluation";
+    static final String GROUP_ID = "org.jbpm.test";
+    static final String VERSION = "1.0.0";
+
+    private KModuleDeploymentUnit unit = null;
+    
+    @Autowired
+    private ProcessService processService;
+    
+    @Autowired
+    private DeploymentService deploymentService;
+    
+    @Autowired
+    private EntityManagerFactory entityManagerFactory;
+    
+    @BeforeClass
+    public static void generalSetup() {
+        KieServices ks = KieServices.Factory.get();
+        ReleaseId releaseId = ks.newReleaseId(GROUP_ID, ARTIFACT_ID, VERSION);
+        File kjar = new File("../kjars/evaluation/jbpm-module.jar");
+        File pom = new File("../kjars/evaluation/pom.xml");
+        MavenRepository repository = getMavenRepository();
+        repository.installArtifact(releaseId, kjar, pom);
+
+    }
+    
+    
+    @Before
+    public void setup() {
+        unit = new KModuleDeploymentUnit(GROUP_ID, ARTIFACT_ID, VERSION, null, null, "PER_PROCESS_INSTANCE");
+        deploymentService.deploy(unit);
+    }
+    
+    @After
+    public void cleanup() {
+
+        deploymentService.undeploy(unit);
+    }
+ 
+    @Test
+    public void testProcessStartAndAbort() {
+        Map<String, Object> parameters = new HashMap<>();
+        parameters.put("employee", "john");
```

Comment:
Just to clarify. If we wanted to use Person entity here (as a process parameter), would we have to include the Person.java class in evaluation kjar as well?

Reply:
no, it's not needed, since this is a custom app and single classpath process can get access to that entity via application classpath and same entity manager can be used without the need to setup new entity manager factory in the jpa marshalling strategy

Reply:
OK, so when is the classpath separate? Only in KIE Server scenario since with KIE Server you communicate remotely? With plain engine (including jUnit engine tests) it is not needed?

Reply:
it's only needed if you don't put this on kie server classpath which usually you don't want to do as this requires unpacking kie server war file.